crates/sui-graphql-rpc/src/config.rs

```diff
@@ -237,6 +246,18 @@ impl ServiceConfig {
         !self.disabled_features.contains(&feature)
     }
 
+    /// List the available versions for this GraphQL service.
+    async fn available_versions(&self, ctx: &Context<'_>) -> Vec<String> {
+        if self.versions.versions.is_empty() {
```

Comment:
Instead of doing this, we could create a `Default` impl for `Versions` that defaults to a singleton array that contains the default version -- that would be a good way to get around the requirement to populate the context with the version.

Also, I noticed that we have another `Version`-related type in this module -- I wonder whether we could re-use that? (e.g. could the config contain a `Vec<Version>`), seems like it would be helpful to ensure that the versions that are configured match the format that we expect versions to be in.

Reply:
Well, we could use the `Version`, but then we'd get all the data. There will be some confusion, because what we accept as a version path is always `year.month`, and not `year.month.patch` or `year.month.patch-sha`, so it might be confusing to see the full version, and it will fail when trying to use that for the path. Showing just available versions with year.month might be reasonable, and the header will give the full version anyway.

If we use `Version`, we'd have to update every single release with the new sha that is being deployed to the config toml file, so that's a bit cumbersome. Probably can be automated but I don't know for sure.

Reply:
The problem with implementing `Default` trait is about reading what the actual version. As `VERSION` is static in main, it is passed as a variable to some function calls, but it cannot be otherwise accessed outside of `main.rs`. That's why I suggested to add it to the ServerConfig, or adding it to the context. I might be wrong here though :D

Reply:
The reason `VERSION` is in `main` is because of the git SHA. But we only want the major and minor version, which we should be able to access using `env!` from here.

Reply:
I see.

Reply:
💡